cli/src/commands/sidekick.rs

```diff
@@ -0,0 +1,197 @@
+			&openapi_config_cloud_unauthed,
+		)
+		.await;
+		if let Err(err) = prepare_res.as_ref() {
+			struct_fmt::print(
+				&Format::Json,
+				&SideKickResponse::Err(SideKickResponseErr::APIError(format!("{err:?}"))),
+			)?;
+		}
+		let prepare_res = prepare_res.context("cloud_devices_links_prepare")?;
+
+		// Open link in browser
+		if webbrowser::open_browser_with_options(
+			webbrowser::Browser::Default,
+			&prepare_res.device_link_url,
+			webbrowser::BrowserOptions::new().with_suppress_output(true),
+		)
+		.is_ok()
+		{
+			term::status::info(
+				"Waiting for link",
+				"Select the game to link in your browser",
+			);
+		} else {
+			// TODO: The case where the browser fails to open is not
+			// handled well from Godot
+			eprintln!(
+				"{}\n  {}",
+				style("Visit the link below").bold().blue(),
+				style(&prepare_res.device_link_url)
+					.italic()
+					.underlined()
+					.cyan()
+			);
+		}
+
+		// Wait for link to complete
+		let mut watch_index = None;
+		let token = loop {
+			let prepare_res = rivet_api::apis::cloud_devices_links_api::cloud_devices_links_get(
+				&openapi_config_cloud_unauthed,
+				&prepare_res.device_link_token,
+				watch_index.as_ref().map(String::as_str),
+			)
+			.await;
+			if let Err(err) = prepare_res.as_ref() {
+				println!("Error: {err:?}");
+			}
+			let prepare_res = prepare_res.context("cloud_devices_links_get")?;
+
+			watch_index = Some(prepare_res.watch.index);
+
+			if let Some(token) = prepare_res.cloud_token {
+				break token;
+			}
+		};
+
+		// Create new context
+		let new_ctx = cli_core::ctx::init(
+			api_endpoint,
+			// Exclude overridden access token to check the token
+			token.clone(),
+		)
+		.await?;
+
+		// Inspect the token
+		let inspect_res =
+			rivet_api::apis::cloud_auth_api::cloud_auth_inspect(&new_ctx.openapi_config_cloud)
+				.await;
+		if let Err(err) = inspect_res.as_ref() {
+			println!("Error: {err:?}");
+		}
+		let inspect_res = inspect_res.context("cloud_auth_inspect")?;
+
+		// Find the game ID
+		let Some(game_cloud) = inspect_res.agent.game_cloud.as_ref() else {
+			bail!("token is not a GameCloud token")
+		};
+		let game_id = game_cloud.game_id;
+
+		// Extract game data
+		let game_res = rivet_api::apis::cloud_games_games_api::cloud_games_games_get_game_by_id(
+			&new_ctx.openapi_config_cloud,
+			&game_id.to_string(),
+			None,
+		)
+		.await;
+		if let Err(err) = game_res.as_ref() {
+			println!("Error: {err:?}");
+		}
+		let game_res = game_res.context("cloud_games_games_get_game_by_id")?;
+		let display_name = game_res.game.display_name;
+
+		// Write the token
+		global_config::mutate_project(|x| x.tokens.cloud = Some(token)).await?;
+
+		term::status::success("Token Saved", display_name);
+
+		struct_fmt::print(
+			&Format::Json,
+			&SideKickResponse::Ok(SideKickResponseOk {
+				messages: vec![SideKickMessage::String("Token Saved".to_string())],
+			}),
+		)?;
```

Comment:
@NathanFlurry I'm trying to come up with a good schema to reply back to callers (Godot, Unity, etc) in a consistent way. Do you have any thoughts on how this would be best handled? The current version I threw together here is likely quite verbose.

An alternative to this is just passing back super loose messages. I don't think there will be a ton coming back from the CLI, though it would be good if it could pass error messages (or a list of errors) in a very readable way.

Reply:
Sounds like you're looking for a serde compatible `Result`? So something like:

```json
{
"ok": { "foo": "bar" }
}
```

And build a standard error format similar to our [APIs](https://github.com/rivet-gg/rivet/blob/8d6a4a98ee645205c20afa7d71f058a84f883f59/lib/api-helper/build/src/error.rs#L12):

```json
{
"err": {
"code":"MY_ERROR",
"message":"my error here",
"documentation": "https://rivet.gg/docs/xxxx",
"metadata": null
}
}
```

Would be interesting to leverage the existing auto-generated error docs for the CLI errors so we can link out to help resolving the error. Would be pretty cool to have the same error system handled all the way from the internal services (already done), to the API (already done), to the CLI.

Reply:
For now, I've changed this to a simple struct:
```
#[derive(Serialize)]
enum SideKickResponse {
	Ok(Value),
	Err(Value),
}
```
However, I believe I saw a small amount into some error handling already in the CLI. It didn't look too usable off the bat, but I'll take a deeper look with the context of the rest of the project.